### PR TITLE
backend/enhancement/dashboard_subscription_api_planResume

### PR DESCRIPTION
**Pull request by Anuragini-2506**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

## Description

Added planResume api in the dashboard ( BPP )

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [x] This PR contains API breaking changes

## How did you test it?

Tested locally

## Checklist

- [ ] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable
- [ ] No leak detected in leakcanary

### CODE REVIEW

File: Backend/app/dashboard/provider-dashboard/src/API/ProviderPlatform/DynamicOfferDriver/Subscription.hs [hunk 1]
```diff
@@ -109,3 +115,10 @@ currentPlan :: ShortId DMerchant.Merchant -> ApiTokenInfo -> Id Common.Driver ->
 currentPlan merchantShortId apiTokenInfo driverId = withFlowHandlerAPI $ do
   checkedMerchantId <- merchantAccessCheck merchantShortId apiTokenInfo.merchant.shortId
   Client.callDriverOfferBPP checkedMerchantId (.subscription.currentPlan) driverId
+
+planResume :: ShortId DMerchant.Merchant -> ApiTokenInfo -> Id Common.Driver -> FlowHandler APISuccess
+planResume merchantShortId apiTokenInfo driverId = withFlowHandlerAPI $ do
+  checkedMerchantId <- merchantAccessCheck merchantShortId apiTokenInfo.merchant.shortId
+  transaction <- buildTransaction SD.SuspendPlanEndpoint apiTokenInfo (Just driverId)
```
Comment (0utkarsh) [inline Backend/app/dashboard/provider-dashboard/src/API/ProviderPlatform/DynamicOfferDriver/Subscription.hs:122]: Incorrect endpoint, please create a new endpoint

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Tools/Notifications.hs [hunk 2]
```diff
@@ -572,6 +572,35 @@ notifyPaymentModeManualOnSuspend merchantId personId mbDeviceToken = do
...
+          fcmEntityIds = personId.getId,
+          fcmEntityData = (),
+          fcmNotificationJSON = FCM.createAndroidNotification title body notifType
+        }
+    title = FCMNotificationTitle "Payment mode changed to Autopay"
+    body =
+      FCMNotificationBody $
+        unwords
+          [ "Your UPI Autopay has been resumed."
+          ]
```
Comment (0utkarsh) [inline Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Tools/Notifications.hs:602]: To do in next iteration : Merge all plan notifs

File: Backend/app/dashboard/provider-dashboard/src/API/ProviderPlatform/DynamicOfferDriver/Subscription.hs [hunk 3]
```diff
@@ -61,6 +62,10 @@ type CurrentPlan =
   ApiAuth 'DRIVER_OFFER_BPP 'SUBSCRIPTION 'CURRENT_PLAN
     :> SD.CurrentPlan
 
+type ResumePlan =
+  ApiAuth 'DRIVER_OFFER_BPP 'SUBSCRIPTION 'CURRENT_PLAN
```
Comment (0utkarsh) [inline Backend/app/dashboard/provider-dashboard/src/API/ProviderPlatform/DynamicOfferDriver/Subscription.hs:66]: Entity should be `RESUME_PLAN`. Please create that entity

Comment (0utkarsh) [reply Backend/app/dashboard/provider-dashboard/src/API/ProviderPlatform/DynamicOfferDriver/Subscription.hs:66]: Can you also write migrations to provide access to these entities?

**Review by 0utkarsh — CHANGES_REQUESTED** [review]

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Tools/Notifications.hs (hunk 2 above)
Comment (Anuragini-2506) [reply Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Tools/Notifications.hs:602]: okay will do that in separate PR

File: Backend/app/dashboard/provider-dashboard/src/API/ProviderPlatform/DynamicOfferDriver/Subscription.hs (hunk 3 above)
Comment (Anuragini-2506) [reply Backend/app/dashboard/provider-dashboard/src/API/ProviderPlatform/DynamicOfferDriver/Subscription.hs:66]: okay

File: Backend/app/dashboard/provider-dashboard/src/API/ProviderPlatform/DynamicOfferDriver/Subscription.hs (hunk 1 above)
Comment (Anuragini-2506) [reply Backend/app/dashboard/provider-dashboard/src/API/ProviderPlatform/DynamicOfferDriver/Subscription.hs]: Added a new endpoint as requested

**Comment by hkmangla** [comment]
Already merged

**Pull request closed without merging**